extensions/ql-vscode/src/data-extensions-editor/generate-flow-model.ts

```diff
@@ -0,0 +1,146 @@
+import { CancellationToken } from "vscode";
+import { DatabaseItem } from "../local-databases";
+import { join } from "path";
+import { QueryRunner } from "../queryRunner";
+import { CodeQLCliServer } from "../cli";
+import { extLogger, TeeLogger } from "../common";
+import { definitions } from "./yaml";
+import { ProgressCallback } from "../progress";
+import { getOnDiskWorkspaceFolders } from "../helpers";
+import { ModeledMethod, ModeledMethodType } from "./modeled-method";
+
+class FlowModelGenerator {
```

Comment:
What's the reasoning creating a class for this? I don't see it's purpose considering we provide a function that takes all the dependencies/args, creates an instance of the function and executes 1 function on it. Apologies if I'm missing something obvious.

Reply:
The reasoning is just to ensure we don't need to pass around all of these dependencies, especially since the same function taking the dependencies is called four times. Putting it in a class encapsulates this nicely.

An alternative which avoids passing these dependencies four times would be creating a nested function that calls the actual function with the correct parameters, but that also doesn't help with readability.

Reply:
Sorry I hadn't spotted that we were calling it four times! I think we've generally used this pattern - creating a class that gets the dependencies in the constructor and then exposing a function to call, so I'm happy leaving it like that.